src/components/RedirectIfAuthenticated.tsx

```diff
@@ -1,11 +1,15 @@
 import React from 'react';
-import { Navigate } from 'react-router-dom';
+import { Navigate, useLocation } from 'react-router-dom';
 import { useAuth } from '../AuthProvider';
 
 const RedirectIfAuthenticated = ({ children }) => {
   const { currentUser } = useAuth();
+  const location = useLocation();
 
-  if (currentUser) {
+  console.log('currentUser:', currentUser);
+  console.log('current path:', location.pathname);
+
+  if (currentUser && location.pathname !== '/registration') {
     return <Navigate to="/character" replace />;
   }
```

Comment:
To trochę pokrętne.
Powinieneś stworzyć komponent ProtectedRoute który opakuje każdy komponent wymagający zalogowania.
W nim sprawdzasz czy user jest zalogowany, jeśli tak to renderujesz mu to co opakowałeś, jeżeli nie, to jest redirect na login.
Dla strony login i register robisz osobne wrappery lub implementujesz logikę w środku tych dwóch podstron, jezeli user jest zalogowany, to robisz redirect z nich na characters, jeżeli nie, to renderujesz formularze.